lib/linter.js

```diff
@@ -309,34 +316,54 @@ export default class Linter {
     }
 
     let hasBOM = options.source.codePointAt(0) === 0xfe_ff;
-    let source = hasBOM ? options.source.slice(1) : options.source;
+    let rawSource = hasBOM ? options.source.slice(1) : options.source;
 
-    let templateAST;
+    let templateInfos = extractTemplates(rawSource, options.filePath);
 
-    try {
-      templateAST = parse(source);
-    } catch (error) {
-      let message = buildErrorMessage(options.filePath, error);
-      results.push(message);
-      return results;
-    }
+    for (let templateInfo of templateInfos) {
+      let source = templateInfo.template;
+      let templateAST;
 
-    let { failures, rules } = this.buildRules({
-      fileConfig,
-      filePath: options.filePath,
-      configResolver: options.configResolver,
-      rawSource: source,
-    });
-    results.push(...failures);
-
-    for (let rule of rules) {
       try {
-        let visitor = await rule.getVisitor();
-        transform(templateAST, () => visitor);
-        results.push(...rule.results);
+        templateAST = parse(source);
```

Comment:
Personally, I'd prefer to correct the location information at this level instead of modifying result info.

Specifically, I'd like to:

1. in the short term, traverse this AST and update all the `.loc` properties to the actually correct values if we are starting "from an offset" (e.g. with `.gjs`/`.gts`)
2. update `@glimmer/syntax` (and `ember-template-recast`) to accept start info for the source string (so that `.loc` info is generated correctly to begin with)

Reply:
I'm fine with this approach, but is it really worth doing another AST traversal just to update the `loc`ation as a temporary solution? I'm not talking in terms of performance, just adding that complexity to the code if it's eventually going to be moved inside `parse`.

Reply:
IMHO, that's totally up to you. If you'd prefer to just do the second option, I'm 100% fine with that.

Reply:
Just for me to clarify, are you fine with this being a TODO (I'm more than happy to tackle this next) or is this a blocker and I should update `parse` first?